### DISCUSSION TITLE
Receiver Board Set for Wltoys A120 A380

### DISCUSSION THREAD

**Discussion by gke** [pascallanger/DIY-Multiprotocol-TX-Module#739] [Q&A] [answered]
Controller boards from AliExpress. https://www.aliexpress.com/item/1005002137064203.html?spm=a2g0o.order_list.0.0.18751802X638ig

I have tried V911S and SHSS with no luck. Before I get into packet tracing has anyone else managed to get these working.

Thanks

**Comment by pascallanger** [comment]
Have you tried the MT99xx subprotocols https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/blob/master/Protocols_Details.md#mt99xx---17 and https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/blob/master/Protocols_Details.md#mt99xx2---92

**Comment by gke** [comment]
Thanks Pascal mt99xx A180 is being detected but not binding completely - no response from controls but LED goes out suggesting it may have bound or almost bound. Possibly channel assignment issue. I will borrow a controller in a few days to check further.

**Reply by pascallanger** [reply]
Have you tried all sub protocols? The MT99xx core is the same but with some variations. So it might be that one start to works but not completely and another will match perfectly fine.

**Comment by gke** [comment]
Yes I worked through all of the variants and the A180 is the only one that seems to be going at least partially through the bind. I can get a throttle response on CH6 but that may be a trim as the A120 is intended for the differential motor steering in the A380. The main LED goes out but that can be in the flags somewhere. The board has LED string drivers but  they are not connected yet just the one motor. 

I will get back to it tomorrow morning and be as systematic as I can. I will also check what RF chip they are using if the codes have not been removed. I have also found the related topic on Deviation.

**Comment by gke** [comment]
I rechecked the MT99X variants and A180 is the only one that binds. I also rechecked V911S GD00X etc with a greater distance between my RadioMaster 16S and the Rx without success.

Using MT99X/A180 I connected two motors and the LED strips. Before I turn on the Tx the motors turn on full then pulse along with the main board LED as the Rx attempts to bind. Once the Tx is turned on auto bind works.

The aileron channel (CH1) changes causes one LED strip to be brighter than the other and vice versa. The throttle (CH3) causes both strips to increase or decrease in intensity. The motors however do not throttle up.  CH6 turns on both motors at about 60% throw. There is no variation in the throttle.

So overall the motors are not arming although the LED strips indicate that the channel assignments are OK. It is possible the gyro/acc initialisation is not being triggered or completing. I will connect my scope to the motor channels next to see if there are any signals.

The transceiver chip is a PAN163CX used on some of the H8 quads and the other device I think is a 16A 769H AA6. The board is marked ZD A120R 2020/8/24. My boards are V2 (Green) of the particular type. 

![IMG_9092](https://user-images.githubusercontent.com/5491685/194178845-d5b241a0-140f-489a-b669-b4620097bbe9.jpg)
![IMG_9093](https://user-images.githubusercontent.com/5491685/194178852-f2e2a7e7-8d50-4cfd-bbbc-72ce3da1b781.jpg)
![IMG_9091](https://user-images.githubusercontent.com/5491685/194178855-15f7b4a5-b743-4d53-bd05-3f19eecfdd8e.jpg)

**Comment by pascallanger** [comment]
What I need is a dump of the original radio using the xn297dump protocol on the multi module running a debug firmware. It's easy if you have an external module but requires siding and a ftdi if you have an internal module.

**Reply by gke** [reply]
Thanks Pascal. 

Yes I have used the dump process before with the FX620 protocol we did  a little while ago. In this case I bought the A120 boards by themselves without the controller but will try to borrow one.

**Comment by gke** [comment]
OK I have the obvious solution. They swapped the connectors used for motors with those used for the LED strips on this version so we can record this as closed but perhaps add A120 to the MT99X A180 as being the same. I have not fulled checked the channel assignements but Ch6 is the LEDs.

**Comment by pascallanger** [comment, marked as answer]
Can you check all the channels assignments so I can update the doc?

**Reply by gke** [reply]
Will do ASAP. The original controller has several momentary switches the states of which are probably being folded into a flags byte.

**Reply by gke** [reply]
I now have a V2 transmitter.

Packet[6] bit0 is RATE and bit 1 is LED. The switches on the transmitter are changeover i.e. click on..click off.

There appears to be no stabiliser disable as previously believed.

So that just leaves channel mapping? I have used channel 5 for RATE and channel 6 for LED.

**Reply by pascallanger** [reply]
Please check if this is correct: https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/blob/master/Protocols_Details.md#sub_protocol-a180---5

**Reply by gke** [reply]
Yes the documentation is correct for what I am using.  

The V2 Tx I have only has control of two bits for RATE and LED and it is possible that the 0x20 ORed in for the A180 is the 3D_6G (rate/angle) control? I will try to check that today.
